src/Covid19Dashboard/IllinoisMapChart/index.jsx

```diff
@@ -67,6 +67,52 @@ class IllinoisMapChart extends React.Component {
       },
       hoverInfo: null,
     };
+    this.mapData = {
+      modeledCountyGeoJson: null,
+      colors: {},
+      colorsAsList: null,
+    };
+  }
+  componentDidUpdate() {
+    if (!(this.mapData.colorsAsList === null
+          && Object.keys(this.props.jsonByLevel.county).length > 0)) {
+      return;
+    }
+    if (Object.keys(this.props.jsonByLevel.country).length && !this.choroCountyGeoJson) {
+      this.choroCountyGeoJson = addDataToGeoJsonBase(
+        this.props.jsonByLevel.county,
+      );
+    }
+    this.mapData.modeledCountyGeoJson = filterCountyGeoJson(this.props.modeledFipsList);
+
+    // Finds max value in data set then finds its base 10 exponent
+    const maxValExponent = Math.log10(this.mapData.modeledCountyGeoJson.features
+      .map(obj => obj.properties.confirmed)
+      .sort((a, b) => b - a)[0]);
+
+    // Math to calculate Index range for map
+    const colorRangeMath = (base) => {
+      // applies maxValExponent to base then rounds down to greatest place
+      const tempNum = Math.ceil(base ** maxValExponent);
+      const roundingDigits = 10 ** (tempNum.toString().length - 1);
+
+      return Math.floor(tempNum / roundingDigits) * roundingDigits;
+    };
+
+    this.mapData.colors = {
+      0: '#FFF',
+      [colorRangeMath(2)]: '#F7F787',
+      [colorRangeMath(3)]: '#EED322',
+      [colorRangeMath(4)]: '#E6B71E',
+      [colorRangeMath(5)]: '#DA9C20',
```

Comment:
why do we use arrays here?

Reply:
it was an artifact of a previous version of the code, ill remove them

Reply:
the array is used to evaluate the function so you can use it as the ID, is there a cleaner way to do this?

Reply:
oh does it not work without the array? 🤯

Reply:
yep does not work without the brackets, uses the brackets as a resolver in this case, kind of odd, otherwise you get typeOf errors

Reply:
interesting